src/NATS.Client.JetStream/INatsJSFetch.cs

```diff
@@ -7,6 +8,11 @@ namespace NATS.Client.JetStream;
 /// </summary>
 public interface INatsJSFetch : IAsyncDisposable
 {
+    /// <summary>
+    /// The reason why the fetch has ended
+    /// </summary>
+    NatsSubEndReason EndReason { get; }
```

Comment:
This isn't something we discussed before, we probably shouldn't expose it for the time being.

Reply:
From my side this was only a proposal. 

For us it's important that we can read all messages as fast as possible. But to read all messages we have to know if there are more messages. The `EndReason` is the flag where we get get the information. If you don't want to expose this info we could cast the `INatsJSFetch` to `NatsSubBase` and get the info from there. But I think this is also not the idea, right?

Reply:
So, `EndReason` is there to give some insight to why subscription (in this case Fetch) was 'terminated'. `NatsSubBase` is a lower level construct so it would be leaking that lower level data to higher level functions having said that I'm not sure how the applications would be using this. Obviously it's helpful to you. Please could you expand a little if possible, how `EndReason` is helping with `Fetch`?

Reply:
Here is a small code example:

```
var fetchOpts = new NatsJSFetchOpts() {Serializer = new RawDataSerializer(), NoWait = true, MaxMsgs = 10};
NatsSubEndReason endReason;
do
{
    await using var sub = await consumer.FetchAsync<RawData>(fetchOpts, cts.Token);
    await foreach (var msg in sub.Msgs.ReadAllAsync(cts.Token))
    {
        await msg.AckAsync(cancellationToken: cts.Token);
        Report(++count, stopwatch, $"data: {msg.Data}");
    }

    endReason = sub.EndReason;
} while (endReason == NatsSubEndReason.MaxMsgs);
```

In our application we have the use case, that we have to read all messages from a dedicated consumer until the end.  The EndReason is very nice to figure it out, if there are more to read or not.

Reply:
Thank you for the sample code @simonhoss 🙏 

Do you think we could translate that for fetch as e.g. `FetchEndReason` since most of the elements on `NatsSubEndReason` would be irrelevant?

@caleblloyd wdyt?

Reply:
For me this is ok. What would be the content of FetchEndReason?

Reply:
As it stands right now, I don't think we return EndReason for anything in the public APIs - Subscriptions, Consumers, Fetch requests, etc.  I don't know that we should necessarily add a 1-off for only Fetch but not the rest of them.